nmigen_boards/dev/spi.py

```diff
@@ -4,16 +4,30 @@
 __all__ = ["SPIResource"]
 
 
-def SPIResource(number, *, cs, clk, mosi, miso, int=None, reset=None, attrs=None):
+def SPIResource(name_or_number, number=None, *, cs, clk, mosi, miso, int=None, reset=None, attrs=None, role="host"):
```

Comment:
This can use `def SPIResource(*args, cs, ...)` for simplicity, with `Resource.family(*args, ...)` below.